### PR TITLE
Cache CR's becomes orphan after cluster deletion

### LINKED ISSUE #450: Cache CR's becomes orphan after cluster deletion
Looks like we need to update Cache CR with Infinispan CR owner reference after Cache CR processing to prevent orphans after Infinispan CR deletion. We should avoid to use finalizers for this use case.

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #450

### CODE REVIEW

**Comment by dmvolod** [comment]
@rigazilla please review, I tried to fix #509, but it still not resolved

File: deploy/cr/auth/cr_minimal_with_auth.yaml [hunk 1]
```diff
@@ -4,5 +4,7 @@ metadata:
   name: example-infinispan
 spec:
   replicas: 2
+  service:
+    type: DataGrid
```
Comment (rigazilla) [inline deploy/cr/auth/cr_minimal_with_auth.yaml:8]: better to create a new file under `deploy/cr/datagrid` instead of changing this

Comment (dmvolod) [reply deploy/cr/auth/cr_minimal_with_auth.yaml:8]: Yes, sure, will create a new one with DataGrid type

**Comment by rigazilla** [comment]
I think we can merge this first. I'll rebase my #515

**Comment by dmvolod** [comment]
> I think we can merge this first. I'll rebase my #515

No, @rigazilla , please merge #515 before, it's not a problem to rebase this PR later

File: pkg/controller/cache/cache_controller.go [hunk 2]
```diff
@@ -176,13 +177,23 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
 				}
 			}
 		}
+
+		ok, err := common.IsOwner(ispnInstance, instance, r.scheme)
```
Comment (rigazilla) [inline pkg/controller/cache/cache_controller.go:181]: It seems that the `SetControllerReference` does some similar checks. Do we need to add our owns?

Comment (dmvolod) [reply pkg/controller/cache/cache_controller.go:181]: Yes, `SetControllerReference` check for already set owner, but this validation prevents a redundant CR update

File: pkg/controller/cache/cache_controller.go [hunk 3]
```diff
@@ -176,13 +177,21 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
 				}
 			}
 		}
+
+		if !metav1.IsControlledBy(instance, ispnInstance) {
```
Comment (dmvolod) [inline pkg/controller/cache/cache_controller.go:181]: Changed to `apimachinery` implementation instead of custom one

**Comment by rigazilla** [comment]
merged, thanks @dmvolod !

**Pull request merged**